projects/engine/src/connector.ts

```diff
@@ -77,11 +78,14 @@ export abstract class PluginConnector extends Plugin {
   protected async handshake() {
     if (!this.loaded) {
       const methods: string[] = await this.callPluginMethod('handshake', [this.profile.name])
-      this.call('manager', 'updateProfile', this.profile)
-      this.loaded = true
       if (methods) {
         this.profile.methods = methods
       }
+      this.call('manager', 'updateProfile', this.profile)
+      this.loaded = true
```

Comment:
If the `loaded` is set at the end (especially after the `await` - line 80), it is probable that `this.callPluginMethod('handshake',  ... ` can be called several times.

Reply:
e.g if 2 handshake are triggered before the execution reach the `this.loaded = true`

Reply:
Ok, but if handshake fails you don't want to set it to loaded...
I don't see the risk of having two handshakes except updating twice the profile methods.